app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,40 @@
+class AccountActivationsController < ApplicationController
+  before_action :load_user, :check_user_exists
+  before_action :check_already_activated
+  before_action :check_authentication
+
+  # GET: /account_activations/:id/edit
+  def edit
+    @user.activate
+    log_in @user
+    flash[:success] = t("account_activations.edit.success")
+    redirect_to @user
+  end
+
+  private
+
+  def load_user
+    @user = User.find_by(email: params[:email])
+  end
+
+  def check_user_exists
+    return if @user
+
+    flash[:danger] = t("account_activations.edit.user_not_found")
+    redirect_to root_url
+  end
```

Comment:
```suggestion
  def load_user
    return if @user = User.find_by(email: params[:email])

    flash[:danger] = t("account_activations.edit.user_not_found")
    redirect_to root_url
  end
```
Sau chú ý các logic load_ này e nha